src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Invitations/InvitationsControllerNegativeTests.cs

```diff
@@ -133,12 +165,536 @@ public async Task CreateInvitation_AsViewer_ShouldReturnForbidden_WhenPermission
+
+        private IEnumerable<ParticipantDtoEdit> ConvertToParticipantDtoEdit(IEnumerable<ParticipantDtoGet> participants)
+        {
+            var editVersionParticipantDtos = new List<ParticipantDtoEdit>();
+            participants.ToList().ForEach(p => editVersionParticipantDtos.Add(
+                new ParticipantDtoEdit
+                {
+                    ExternalEmail = p.ExternalEmail,
+                    FunctionalRole = p.FunctionalRole,
+                    Organization = p.Organization,
+                    Person = p.Person?.Person,
+                    SortKey = p.SortKey
+                }));
+
+            return editVersionParticipantDtos;
+        }
         #endregion
 
-        #region Edit (Sign)
-        //todo add tests 
+        #region Sign 
+        [TestMethod]
+        public async Task SignPunchOut_AsAnonymous_ShouldReturnUnauthorized()
+            => await InvitationsControllerTestsHelper.SignPunchOutAsync(
+                UserType.Anonymous,
+                TestFactory.PlantWithoutAccess,
+                9999,
+                _sigurdSigner.Id,
+                _sigurdSigner.RowVersion,
+                HttpStatusCode.Unauthorized);
+
+        [TestMethod]
+        public async Task SignPunchOut_AsHacker_ShouldReturnBadRequest_WhenUnknownPlant()
+            => await InvitationsControllerTestsHelper.SignPunchOutAsync(
+                UserType.Hacker,
+                TestFactory.UnknownPlant,
+                9999,
+                _sigurdSigner.Id,
+                _sigurdSigner.RowVersion,
+                HttpStatusCode.BadRequest,
+                "is not a valid plant");
 
+        [TestMethod]
+        public async Task SignPunchOut_AsPlanner_ShouldReturnBadRequest_WhenUnknownPlant()
+            => await InvitationsControllerTestsHelper.SignPunchOutAsync(
+                UserType.Planner,
+                TestFactory.UnknownPlant,
+                9999,
+                _sigurdSigner.Id,
+                _sigurdSigner.RowVersion,
+                HttpStatusCode.BadRequest,
+                "is not a valid plant");
+
+        [TestMethod]
+        public async Task SignPunchOut_AsHacker_ShouldReturnForbidden_WhenPermissionMissing()
+            => await InvitationsControllerTestsHelper.SignPunchOutAsync(
+                UserType.Hacker,
+                TestFactory.PlantWithAccess,
+                9999,
+                _sigurdSigner.Id,
+                _sigurdSigner.RowVersion,
+                HttpStatusCode.Forbidden);
+
+        [TestMethod]
+        public async Task SignPunchOut_AsPlanner_ShouldReturnBadRequest_WhenUnknownInvitationId() 
+            => await InvitationsControllerTestsHelper.SignPunchOutAsync(
+                UserType.Planner,
+                TestFactory.PlantWithAccess,
+                38934,
+                _sigurdSigner.Id,
+                _sigurdSigner.RowVersion,
+                HttpStatusCode.BadRequest);
+        #endregion
+
+        #region Complete
```

Comment:
I think more negative tests regarding permissions should be added:
* Complete should fail if planner (after removing SIGN from planners permissions list)
* Complete should fail if signer (after removing WRITE from signers permissions list)